### DISCUSSION TITLE
DirectoryScanJob for SFTP

### DISCUSSION THREAD

**Discussion by ManuelLopezNavarrete** [quartznet/quartznet#2117] [Ideas] [answered]
Hi,

I have the need to check some folder in a SFTP, what i saw the DirectoryScanJob don't support this, only locals/intranet folders.
I think it will be a nice added functionality. I never do a contribution in an open source project, so sorry in advance if I'm missing some etiquette rule.

I was thinking in do the change in DirectoryScanJob, i don't knows if there any special politic to use some package. For example for this case i was thinking in SSH.NET.
And add a new JobDataMap like:

LOCATION_FOLDER: local (default), SFTP
IP
USER
PASSWORD

Or it will be better implement a new full class like SftpDirectoryScanJob?

**Comment by lahma** [comment, marked as answer]
Thank you for your interest in this project and for your willingness to help out. I think it would probably be easiest to create an entirely new type for the implementation. You could dog-food it as part of your own code base and hust configure it for Quartz to find.

When it seems stable and fulfills your needs, you could propose a PR against Quartz repository.

**Comment by ManuelLopezNavarrete** [comment]
Wonderful, i will start soon as posible .

Thanks
